runtimes/talos/src/main/java/com/talosvfx/talos/runtime/scene/components/PathRendererComponent.java

```diff
@@ -0,0 +1,335 @@
+                Vector2 curr = bezier.valueAt(tmp2, t);
+                points[index++] = curr.cpy();
+            }
+            Vector2 curr = bezier.valueAt(tmp2, 1f);
+            points[index++] = curr.cpy();
+        }
+
+        edgePoints = computeEdgePoints(points, thickness);
+
+
+        int ATTRIBUTE_COUNT = 5;
+        vertices = new float[edgePoints.length * ATTRIBUTE_COUNT + (repeatCount) * 2 * ATTRIBUTE_COUNT];
+
+        float length = 0;
+        Vector2 prev = new Vector2(points[0].x, points[0].y);
+        for (int i = 1; i < points.length; i++) {
+            length += points[i].dst(prev);
+            prev.set(points[i]);
+        }
+
+        float pixelSize = repeatCount / length;
+
+        int idx = 0;
+        this.prev = null;
+        progress = 0;
+        for (int i = 0; i < points.length; i++) {
+            idx = setData(idx, edgePoints[i * 2].x, edgePoints[i * 2].y, edgePoints[i * 2 + 1].x, edgePoints[i * 2 + 1].y, points[i].x, points[i].y, pixelSize);
+
+        }
+
+        int indicesLength = (vertices.length / ATTRIBUTE_COUNT - 2) * 3;
+        indices = new short[indicesLength];
+
+        int tri = 0;
+        for (int i = 0; i < idx / ATTRIBUTE_COUNT - 2; i += 2) {
+            indices[tri++] = (short) i;
+            indices[tri++] = (short) (i + 1);
+            indices[tri++] = (short) (i + 2);
+            indices[tri++] = (short) (i + 1);
+            indices[tri++] = (short) (i + 2);
+            indices[tri++] = (short) (i + 3);
+        }
+    }
+
+    public int getNumSegments() {
+        return controlPoints.length / 3;
+    }
+
+    Vector2[] computeEdgePoints(Vector2[] curvePoints, float offsetDistance) {
+        Vector2[] offsetPoints = new Vector2[curvePoints.length * 2];
+
+        Vector2 firstTangent = getTangent(curvePoints[1], curvePoints[0]);
+        offsetPoints[0] = curvePoints[0].cpy().add(new Vector2(-firstTangent.y, firstTangent.x).nor().scl(offsetDistance));
+        offsetPoints[1] = curvePoints[0].cpy().add(new Vector2(-firstTangent.y, firstTangent.x).nor().scl(-offsetDistance));
+
+        int length = curvePoints.length;
+        Vector2 lastTangent = getTangent(curvePoints[length - 1], curvePoints[length - 2]);
+        offsetPoints[offsetPoints.length - 2] = curvePoints[length - 1].cpy().add(new Vector2(-lastTangent.y, lastTangent.x).nor().scl(offsetDistance));
+        offsetPoints[offsetPoints.length - 1] = curvePoints[length - 1].cpy().add(new Vector2(-lastTangent.y, lastTangent.x).nor().scl(-offsetDistance));
+
+        int idx = 2;
+        for (int i = 1; i < curvePoints.length - 1; i++) {
+            Vector2 p = curvePoints[i];
+            Vector2 t = getTangent(curvePoints[i + 1], curvePoints[i - 1]);
+            Vector2 n = new Vector2(-t.y, t.x).nor().scl(offsetDistance);
+            offsetPoints[idx++] = p.cpy().add(n);
+            offsetPoints[idx++] = p.cpy().add(new Vector2(-t.y, t.x).nor().scl(-offsetDistance));
```

Comment:
same for here, cpy is gonna be generating a lot of garbage, as well as the New VEctors